### PR TITLE
Guides pro

### PR DESCRIPTION
**Pull request by PhilippeR26**
## Motivation and Resolution
Implementation of full guides
...

### RPC version (if applicable)

...

## Usage related changes

- Change 1.
- ...

## Development related changes

- Change 1.
- ...

## Checklist:

- [ ] Performed a self-review of the code
- [ ] Rebased to the last commit of the target branch (or merged it into my branch)
- [x ] Linked the issues which this PR resolves
- [ ] Documented the changes
- [x ] Updated the docs (www)
- [ ] Updated the tests
- [ ] All tests are passing

### CODE REVIEW

**Review by tabaktoni — APPROVED** [review]
"in god we trust"

**Comment by PhilippeR26** [comment]
In the Deploy preview, I noted : 
- define_call_message.md : the table is not shown properly .
- create_account.md : create ArgentX : code not properly displayed.
All is displayed properly in GitHub and in my local md editor.
If somebody know how to correct this ...

File: www/guides/create_account.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+```
+If you want a specific private key, replace `stark.randomAddress()` by your choice.
+
+Then you have to fund this address.  
+How to proceed is out of the scope of this guide, by you can for example :
+- transfert ETH from an other wallet.
+- Bridge ETH to this Starknet address.
+- Use a faucet.
+- Mint ETH on starknet-devnet.
```
Comment (ivpavici) [inline www/guides/create_account.md:53]: I would maybe expand this part a bit, like from:
https://github.com/0xs34n/starknet.js/pull/480/files#diff-c469c62c8cbadad5458c58336bc2394c2c7e8e25bcf1fea0c652bc2deccdb323L83

File: www/guides/create_account.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+// new account abstraction :
+    // Generate public and private key pair.
+const AAprivateKey = stark.randomAddress();
+console.log('New account :\nprivateKey=', AAprivateKey);
+const AAstarkKeyPair = ec.getKeyPair(AAprivateKey);
+const AAstarkKeyPub = ec.getStarkKey(AAstarkKeyPair);
+console.log('publicKey=', AAstarkKeyPub);
+
+// declare the contract
+const compiledAAaccount = json.parse(fs.readFileSync("./compiled_contracts/myAccountAbstraction.json").toString("ascii")
```
Comment (ivpavici) [inline www/guides/create_account.md:150]: Could I ask you to add a section about clarifying this `json.parse` somewhere?
It created a confusion in this issue: https://github.com/0xs34n/starknet.js/issues/473#issuecomment-1371931040

**Comment by ivpavici** [comment]
Also, please @PhilippeR26 , there is a small conflict because of some merges in erc20.md that happened in the meantime!

File: www/guides/connect_contract.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+# 🔌 Connect a deployed contract :
+Once your provider initialized, you can connect a contract already deployed in the network.  
+You need 2 data :
+- the address of the contract
+- the ABI file of the contract (or the compiled contract file, that includes the abi)  
+>If you don't have the abi file, the `provider.getClassAt()` command can help you. When possible, prefer to read the compiled contract from a local Json file, as it's much more faster.
+
+```typescript
+import { Provider, Contract, json } from "starknet";
+```
```
[the remark this answers is not in the document]
Comment (PhilippeR26) [reply www/guides/connect_contract.md:13]: OK

File: www/guides/connect_contract.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
+---
+sidebar_position: 5
+---
+# 🔌 Connect a deployed contract :
```
[the remark this answers is not in the document]
Comment (PhilippeR26) [reply www/guides/connect_contract.md:4]: OK

File: www/guides/L1message.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
+---
+sidebar_position: 13
+---
+# Messages with L1 network
```
[the remark this answers is not in the document]
Comment (PhilippeR26) [reply www/guides/L1message.md:4]: OK

File: www/guides/estimate_fees.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+```typescript
+const { suggestedMaxFee: estimatedFee1 } = await account0.estimateDeclareFee({ 
+	contract: compiledTest, 
+	classHash: testClassHash });
+const declareResponse = await account0.declare({ 
+	contract: compiledTest, 
+	classHash: testClassHash 
+	}, { 
+	maxFee: estimatedFee1.mul(new BN(11)).div(new BN(10)) 
+	});
```
[the remark this answers is not in the document]
Comment (PhilippeR26) [reply www/guides/estimate_fees.md:65]: OK.

File: www/guides/L1message.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+      This function is payable, the payed amount is the message fee.
+      Returns the hash of the message and the nonce of the message.
+    */
+    function sendMessageToL2(
+        uint256 toAddress,
+        uint256 selector,
+        uint256[] calldata payload
+    ) external payable returns (bytes32, uint256);
+
+    /**
```
[the remark this answers is not in the document]
Comment (PhilippeR26) [reply www/guides/L1message.md:29]: Removed.

File: www/guides/L1message.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+## L2 ➡️ L1 messages
+To send a message to L1, you will just invoke a cairo contract function, paying a fee that will pay all the process (in L1 & L2).  
+If necessary you can estimate this fee with the generic `estimateInvokeFee` function :
+```typescript
+const { suggestedMaxFee: estimatedFee1 } = await account0.estimateInvokeFee({ 
+	contractAddress: testAddress, 
+	entrypoint: "withdraw_to_L1", 
+	calldata: ["123456789", "30"] });
+```
+The result is in estimatedFee1, of type BN.
```
[the remark this answers is not in the document]
Comment (PhilippeR26) [reply www/guides/L1message.md:53]: OK

**Comment by PhilippeR26** [comment]
> Also, please @PhilippeR26 , there is a small conflict because of some merges in erc20.md that happened in the meantime!

My  erc20.md is replacing completely the old one.

File: www/guides/create_account.md (hunk 1 above)
Comment (PhilippeR26) [reply www/guides/create_account.md:53]: OK

File: www/guides/create_account.md (hunk 2 above)
Comment (PhilippeR26) [reply www/guides/create_account.md:150]: I already made a paragraph in connect_contract.md : https://github.com/PhilippeR26/starknet.js/blob/guidesPRO/www/guides/connect_contract.md#get-the-abi-from-a-compiled-file-
I will add something in create_account.md.

Comment (PhilippeR26) [reply www/guides/create_account.md:150]: Done.

File: www/guides/create_account.md (hunk 1 above)
Comment (ivpavici) [reply www/guides/create_account.md:61]: Oh, this link I made doesn't make sense anymore... suggestion what I meant:
```suggestion
- transfert ETH from an other wallet.
- Bridge ETH to this Starknet address.
- Use a faucet. (https://faucet.goerli.starknet.io/)
- Mint ETH on starknet-devnet, like so:
```bash
curl -X POST http://127.0.0.1:5050/mint -d '{"address":"0x04a093c37ab61065d001550089b1089922212c60b34e662bb14f2f91faee2979","amount":50000000000000000000,"lite":true}' -H "Content-Type:application/json"
// {"new_balance":50000000000000000000,"tx_hash":null,"unit":"wei"}
```
```

**Comment by ivpavici** [comment]
> > Also, please @PhilippeR26 , there is a small conflict because of some merges in erc20.md that happened in the meantime!
> 
> My erc20.md is replacing completely the old one.

Yeah, but it seems like git still sees this like a conflict :(
image

**Comment by PhilippeR26** [comment]
The current `erc20.md` has to be deleted.
The subject of ERC20 is now located in  `use_ERC20.md`.
I don't know how to manage this point in GitHub.

File: www/guides/estimate_fees.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (PhilippeR26) [reply www/guides/estimate_fees.md]: Done.

**Pull request closed without merging**